sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableChange.java

```diff
@@ -726,7 +746,16 @@ public String[] fieldNames() {
      * is not provided as updating column default values does not need to back-fill existing data.
      * Empty string means dropping the column default value.
      */
-    public String newDefaultValue() { return newDefaultValue; }
+    public String newDefaultValue() {
+      return newDefaultValue == null ? "" : newDefaultValue.getSql();
+    }
+
+    /**
+     * Returns the column default value as {@link DefaultValue}. The default value literal
+     * is not provided as updating column default values does not need to back-fill existing data.
+     * Empty string and Null literal means dropping the column default value.
+     */
+    public DefaultValue newModelDefaultValue() { return newDefaultValue; }
```

Comment:
I am not sure I like the name here. I get that the ideal name `newDefaultValue` is already taken, but let's brainstorm a bit more. What about any of these?
- `newValue()`
- `updatedDefaultValue()`
- `newColumnDefaultValue()`
- `newCurrentDefaultValue()`

Any other suggestions, @szehon-ho @cloud-fan @gengliangwang?

Reply:
yea agree its not the greatest name.  Renamed it newCurrentDefault, let me know if strong opinion otherwise.